static/app/components/events/eventStatisticalDetector/aggregateSpanOps/pieChart.tsx

```diff
@@ -0,0 +1,200 @@
+import {Component, createRef} from 'react';
+import {Theme, withTheme} from '@emotion/react';
+import type {PieSeriesOption} from 'echarts';
+
+import BaseChart, {BaseChartProps} from 'sentry/components/charts/baseChart';
+import Legend from 'sentry/components/charts/components/legend';
+import PieSeries from 'sentry/components/charts/series/pieSeries';
+import {ReactEchartsRef, Series} from 'sentry/types/echarts';
+import {formatPercentage, getDuration} from 'sentry/utils/formatters';
+
+export interface PieChartSeries
+  extends Series,
+    Omit<PieSeriesOption, 'id' | 'color' | 'data'> {}
+
+interface Props extends Omit<BaseChartProps, 'series'> {
+  // TODO improve type
+  data: any;
+  series: PieChartSeries[];
+  theme: Theme;
+  selectOnRender?: boolean;
+}
+
+class PieChart extends Component<Props> {
+  componentDidMount() {
+    const {selectOnRender} = this.props;
+
+    if (!selectOnRender) {
+      return;
+    }
+
+    // Timeout is because we need to wait for rendering animation to complete
+    // And I haven't found a callback for this
+    this.highlightTimeout = window.setTimeout(() => this.highlight(0), 1000);
+  }
+
+  componentWillUnmount() {
+    window.clearTimeout(this.highlightTimeout);
+  }
+
+  highlightTimeout: number | undefined = undefined;
+  isInitialSelected = true;
+  selected = 0;
+  chart = createRef<ReactEchartsRef>();
+
+  // Select a series to highlight (e.g. shows details of series)
+  // This is the same event as when you hover over a series in the chart
+  highlight = dataIndex => {
+    if (!this.chart.current) {
+      return;
+    }
+    this.chart.current.getEchartsInstance().dispatchAction({
+      type: 'highlight',
+      seriesIndex: 0,
+      dataIndex,
+    });
+  };
+
+  // Opposite of `highlight`
+  downplay = dataIndex => {
+    if (!this.chart.current) {
+      return;
+    }
+
+    this.chart.current.getEchartsInstance().dispatchAction({
+      type: 'downplay',
+      seriesIndex: 0,
+      dataIndex,
+    });
+  };
+
+  // echarts Legend does not have access to percentages (but tooltip does :/)
+  getSeriesPercentages = (series: PieChartSeries) => {
+    const total = series.data.reduce((acc, {value}) => acc + value, 0);
+    return series.data
+      .map(({name, value}) => [name, Math.round((value / total) * 10000) / 100])
+      .reduce(
+        (acc, [name, value]) => ({
+          ...acc,
+          [name]: value,
+        }),
+        {}
+      );
+  };
+
+  render() {
+    const {series, theme, ...props} = this.props;
+    if (!series || !series.length) {
+      return null;
+    }
+    if (series.length > 1) {
+      // eslint-disable-next-line no-console
+      console.warn('PieChart only uses the first series!');
+    }
+
+    // Note, we only take the first series unit!
+    const [firstSeries] = series;
+
+    return (
+      <BaseChart
+        ref={this.chart}
+        colors={[...theme.charts.getColorPalette(5).reverse()]}
```

Comment:
Okay I'm not 100% sure, but I think this might explain why the colours are switching at some point. 🤔 
I wonder if it's doing an in-place reverse to the colour palette theme and it happens twice after something re-renders. If that's right, then we should do something like this instead, i.e. apply the reverse on the new array

```suggestion
        colors={[...theme.charts.getColorPalette(5)].reverse()}
```

Reply:
Yeah I just semi-verified that's the fix. I alerted inside the `getColorPalette` call and there's something that triggers this to re-render/run again, and right after I dismiss the alert, the colours flip.